### PR TITLE
Implement Choice Tile

### PR DESCRIPTION
**Pull request by michaelmang**
https://helixdesignsystem.github.io/helix-ui/components/choice-tile/

Screen Shot 2020-05-18 at 5 04 05 PM

### CODE REVIEW

File: src/Choice Tile/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  subdued,
+  title,
+  ...rest
+}) => {
+  return (
+    <label className="hxChoice">
+      <input
+        checked={wcBool(checked)}
+        disabled={wcBool(disabled)}
+        invalid={wcBool(invalid)?.toString()}
```
Comment (nicko-winner) [inline src/Choice Tile/index.js:27]:
```suggestion
        checked={checked}
        disabled={disabled}
        invalid={invalid?.toString()}
```
might not need `wcBool` on elements that are not web components.

File: src/Choice Tile/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        name={id}
+        type="radio"
+      />
+      <hx-tile
+        class={classNames({ className, hxSubdued: wcBool(subdued), [SIZES[size]]: true })}
```
Comment (nicko-winner) [inline src/Choice Tile/index.js:32]: I wonder if className should go on the most parent element `label`, (would give you more power over the styles)

File: src/Choice Tile/index.js (hunk 1 above)
Comment (michaelmang) [reply src/Choice Tile/index.js:27]: While testing this out, the checked behavior only worked when using `wcBool`

Comment (nicko-winner) [reply src/Choice Tile/index.js:27]: Ok. Thanks for clarifying.

File: src/Choice Tile/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  title,
+  ...rest
+}) => {
+  return (
+    <label className={classNames({ hxChoice: true, className })}>
+      <input
+        checked={wcBool(checked)}
+        disabled={wcBool(disabled)}
+        invalid={wcBool(invalid)?.toString()}
+        name={id}
```
Comment (nicko-winner) [inline src/Choice Tile/index.js:28]: I wonder if we should point on the field name explicitly with its own prop `name`,  would make using it closer to a native html5 element.  `<ChoiceTile name="foo" /> <input name="bar" />`

File: src/Choice Tile/index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  ...rest
+}) => {
+  return (
+    <label className={classNames({ hxChoice: true, className })}>
+      <input
+        checked={wcBool(checked)}
+        disabled={disabled}
+        invalid={invalid?.toString()}
+        name={id}
+        type="radio"
```
Comment (nicko-winner) [inline src/Choice Tile/index.js:29]: Might need to also pass in `onChange` click handler to listen for when there is a change to the choice title.

File: src/Choice Tile/index.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  id,
+  invalid,
+  onChange,
+  size,
+  subdued,
+  title,
+  ...rest
+}) => {
+  return (
+    <label className={classNames({ hxChoice: true, className })}>
```
Comment (nicko-winner) [inline src/Choice Tile/index.js:23]: non blocking: a potential shorter way to say the same thing:
```suggestion
    <label className={classNames( 'hxChoice', className)}>
```

Comment (michaelmang) [reply src/Choice Tile/index.js:23]: I think this alternative would add a class name of undefined when there is no `className`, right?

Comment (nicko-winner) [reply src/Choice Tile/index.js:23]: only if it was written like this: `classNames={`hxChoice  ${className}`}` but `className()` does not include undefined values. 

from the docs: https://github.com/JedWatson/classnames#readme
```
// other falsy values are just ignored
classNames(null, false, 'bar', undefined, 0, 1, { baz: null }, ''); // => 'bar 1'
```

**Pull request merged**